src/utils/responsiveSize.js

```diff
@@ -0,0 +1,61 @@
+import { css } from "styled-components";
+
+import { typography } from "../theme";
+import {
+  smallAndUp,
+  mediumAndUp,
+  largeAndUp,
+  xLargeAndUp
+} from "../theme/mediaQueries";
+
+export const responsiveSizeMixin = size => css`
+  font-size: ${typography.size[
+    size.constructor === String ? size : size.xSmall
+  ]};
+
+  ${size.small &&
+    smallAndUp`
+    font-size: ${typography.size[size.small]};
+  `};
+
+  ${size.medium &&
+    mediumAndUp`
+    font-size: ${typography.size[size.medium]};
+  `};
+
+  ${size.large &&
+    largeAndUp`
+    font-size: ${typography.size[size.large]};
+  `};
+
+  ${size.xLarge &&
+    xLargeAndUp`
+    font-size: ${typography.size[size.xLarge]};
+  `};
+`;
+
+export const getResponsiveSize = ({ size = "hecto", responsiveSize = null }) =>
+  responsiveSize
+    ? {
+        xSmall: responsiveSize.xSmall || responsiveSize.small || size,
+        small: responsiveSize.small || responsiveSize.xSmall || size,
+        medium:
+          responsiveSize.medium ||
+          responsiveSize.small ||
+          responsiveSize.xSmall ||
+          size,
+        large:
+          responsiveSize.large ||
+          responsiveSize.medium ||
+          responsiveSize.small ||
+          responsiveSize.xSmall ||
+          size,
+        xLarge:
+          responsiveSize.xLarge ||
+          responsiveSize.large ||
+          responsiveSize.medium ||
+          responsiveSize.small ||
+          responsiveSize.xSmall ||
+          size
+      }
+    : size;
```

Comment:
I am not sure we even need this logic anymore. Given our media queries are going from smallest to largest, we don't need to handle it in the code. Because if `xLarge` is not defined, but `xSmall` is, the xSmall will take effect as the mixin would have generated this css:
```
font-size: {xSmall};
```

And in the other case, if a size was defined for larger breakpoint it would result in:
```
font-size: {xSmall};

@media screen and (min-width: {xLarge)) {
    font-size: {xLarge};
}
```

Reply:
👍